File: sample/src/main/java/com/ivianuu/rxserviceconnection/sample/DummyService.java
package com.ivianuu.rxserviceconnection.sample;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.support.annotation.IntDef;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.util.Log;

import com.ivianuu.rxserviceconnection.RxBinder;

import java.util.concurrent.TimeUnit;

import io.reactivex.Observable;

/**
 * @author Manuel Wrage (IVIanuu)
 */
public class DummyService extends Service {

    private RxBinder<DummyService> binder = new RxBinder<DummyService>() {
        @NonNull
        @Override
        public DummyService getService() {
            return DummyService.this;
        }
    };

    public Observable<Long> test() {
        return Observable.interval(1000, TimeUnit.MILLISECONDS);
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return binder;
    }
}

File: sample/src/main/java/com/ivianuu/rxserviceconnection/sample/DummyService.kt
package com.ivianuu.rxserviceconnection.sample

import android.app.Service
import android.content.Intent
import android.os.IBinder
import android.support.annotation.IntDef
import android.util.Log

import com.ivianuu.rxserviceconnection.RxBinder

import java.util.concurrent.TimeUnit

import io.reactivex.Observable

/**
 * Dummy service
 */
class DummyService : Service() {

    private val binder = object : RxBinder<DummyService>() {
        override val service: DummyService
            get() = this@DummyService
    }

    fun test(): Observable<Long> {
        return Observable.interval(1000, TimeUnit.MILLISECONDS)
    }

    override fun onBind(intent: Intent): IBinder? {
        return binder
    }
}
